### PR TITLE
(SIMP-2180) SIMP 6 Rewrite

### PR DESCRIPTION
**Pull request by jeannegreulich**
updated openldap for global catalysts
  removed NSCD calls

SIMP-2180 #close
SIMP-2313 #close
SIMP-1687 #comment pupmod-simp-openldap updated.

### CODE REVIEW

File: metadata.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "simp-openldap",
-  "version": "5.0.1",
+  "version": "6.0.1",
```
Comment (jeefberkey) [inline metadata.json]: 6.0.0

File: manifests/pam.pp [hunk 2]
```diff
@@ -1,374 +0,0 @@
...
-      require => File[$ldap_conf]
-    }
-  }
-
-  if $ssl and $use_simp_pki {
-    Class['pki'] -> File[$ldap_conf]
-  }
-
-  # Complete hackery to account for legacy upgrades.
-  if $use_sssd {
```
Comment (jeefberkey) [inline manifests/pam.pp]: Is this still needed?

Comment (jeannegreulich) [inline manifests/pam.pp]: Yes.  I asked if we needed to check for sssd and was told yes, someone might not want to use it, so leave the checks in.

I changed the name to sssd to match the global catalyst.

File: manifests/server/conf.pp [hunk 3]
```diff
@@ -335,16 +341,16 @@
...
-      $modulepath = ['/usr/lib/openldap']
+    $modulepath = ['/usr/lib/openldap']
   }
 
   if $force_log_quick_kill {
-    include '::incron'
+    include '::simplib::incron'
 
-    incron::system_table { 'nuke_openldap_log_files':
+    simplib::incron::add_system_table { 'nuke_openldap_log_files':
```
Comment (jeefberkey) [inline manifests/server/conf.pp]: These should be changed back

File: manifests/server/syncrepl.pp [hunk 4]
```diff
@@ -24,11 +24,11 @@
   String     $schemachecking    = 'off',
   String     $starttls          = 'critical',
   String     $bindmethod        = 'simple',
-  String     $binddn            = simplib::lookup('simp_options::ldap::bind_dn', { 'default_value' => "${::openldap::base_dn}" }),
+  String     $binddn            = simplib::lookup('simp_options::ldap::sync_dn', {'default_value' => '' }),
```
Comment (jeefberkey) [inline manifests/server/syncrepl.pp]: bind_dn looks up sync_dn?

**Comment by trevor-vaughan** [comment]
The parameters need to be validated against the templates and direct calls to hiera need to be eliminated.

**Comment by jeefberkey** [comment]
Working on fixing the acceptance tests

**Comment by trevor-vaughan** [comment]
So, it *looks* like the tests just need to be updated and I think I'm closing in on that.

Fun fact: Openldap starts even if you require TLS but it doesn't have any certs present :rage:

**Comment by nick-markowski** [comment]
Ref SIMP-2742, SIMP-2447 in merge

**Pull request merged**